src/unsupervised.ts

```diff
@@ -42,15 +46,80 @@ export function registerUnsupervisedActions() {
  */
 export function registerUnsupervisedHandlers() {
     NEURO.client?.onAction((actionData: ActionData) => {
-        if(actionKeys.includes(actionData.name)) {
-            NEURO.actionHandled = true;
-            const result = neuroActionHandlers[actionData.name](actionData);
-            NEURO.client?.sendActionResult(actionData.id, result.success, result.message);
-        }
-        else if(NEURO.tasks.find(task => task.id === actionData.name)) {
+        if (actionKeys.includes(actionData.name) || NEURO.tasks.find(task => task.id === actionData.name)) {
             NEURO.actionHandled = true;
-            const result = handleRunTask(actionData);
-            NEURO.client?.sendActionResult(actionData.id, result.success, result.message);
+
+            let action: ActionWithHandler;
+            if (actionKeys.includes(actionData.name)) {
+                action = neuroActions[actionData.name];
+            }
+            else {
+                const task = NEURO.tasks.find(task => task.id === actionData.name)!;
+                action = {
+                    name: task.id,
+                    description: task.description,
+                    permissions: [PERMISSIONS.runTasks],
+                    handler: handleRunTask,
+                    promptGenerator: () => `Neuro wants to run the task "${task.id}".`,
+                };
+            }
+
+            const effectivePermission = action.permissions.length > 0 ? getPermissionLevel(...action.permissions) : action.defaultPermission ?? PermissionLevel.COPILOT;
+            if (effectivePermission === PermissionLevel.OFF) {
+                NEURO.client?.sendActionResult(actionData.id, true, "Action failed: You don't have permission to execute this action.");
```

Comment:
Use the infinitive of the permission to give more detailed feedback.
```suggestion
                const offPermission = action.permissions.find(permission => getPermissionLevel(permission) === PermissionLevel.OFF);
                NEURO.client?.sendActionResult(actionData.id, true, `Action failed: You don't have permission to ${offPermission?.infinitive ?? 'execute this action'}.`);
```